apis/serving/v1alpha1/servingruntime_webhook.go

```diff
@@ -0,0 +1,214 @@
+//Copyright 2021 IBM Corporation
+//
+//Licensed under the Apache License, Version 2.0 (the "License");
+//you may not use this file except in compliance with the License.
+//You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+//Unless required by applicable law or agreed to in writing, software
+//distributed under the License is distributed on an "AS IS" BASIS,
+//WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//See the License for the specific language governing permissions and
+//limitations under the License.
+//
+package v1alpha1
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	kservev1alpha "github.com/kserve/kserve/pkg/apis/serving/v1alpha1"
+	"github.com/kserve/kserve/pkg/constants"
+	"github.com/kserve/modelmesh-serving/controllers/autoscaler"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+//+kubebuilder:webhook:path=/validate-serving-modelmesh-io-v1alpha1-servingruntime,mutating=false,failurePolicy=fail,sideEffects=None,groups=serving.kserve.io,resources=servingruntimes;clusterservingruntimes,verbs=create;update,versions=v1alpha1,name=servingruntime.modelmesh-webhook-server.default,admissionReviewVersions=v1
+type ServingRuntimeWebhook struct {
+	Client  client.Client
+	decoder *admission.Decoder
+}
+
+func (s *ServingRuntimeWebhook) Handle(ctx context.Context, req admission.Request) admission.Response {
+	var srAnnotations map[string]string
+	srReplicas := uint16(65535)
```

Comment:
just out of curiosity, why set the default replicas value to `65535` rather than `-1` here?

Reply:
It is an unsigned int16 type so -1 is not allowed to use.

Reply:
Since numbers are prone to typos, instead of `uint16(65535)` I think we should rather use a constant like `math.MaxUint16`

https://www.includehelp.com/golang/math-maxuint16-constant-with-examples.aspx

Reply:
Agree and I updated it.

Reply:
Thanks @Jooho